src/main/java/net/snowflake/client/jdbc/SnowflakeConnectString.java

```diff
@@ -130,18 +130,17 @@ public static SnowflakeConnectString parse(String url, Properties info) {
         // if it's a global url
         parameters.put("ACCOUNT", account);
       }
-
-      if (account.contains("_")
-          && parameters.containsKey(
-              SFSessionProperty.ALLOW_UNDERSCORES_IN_HOST.getPropertyKey().toUpperCase())
-          && "false"
-              .equalsIgnoreCase(
-                  (String)
-                      parameters.get(
-                          SFSessionProperty.ALLOW_UNDERSCORES_IN_HOST
-                              .getPropertyKey()
-                              .toUpperCase()))
-          && host.startsWith(account)) {
+      boolean allowUnderscoresInHost = false;
+      if ("true"
+          .equalsIgnoreCase(
```

Comment:
will this cause NPE if this property not present in the parameters?

Reply:
@sfc-gh-igarish no, the test has a case covering this: https://github.com/snowflakedb/snowflake-jdbc/blob/9a39c5a793cbaab0ee45b5b465998fa2c1e33bb6/src/test/java/net/snowflake/client/jdbc/ConnectStringParseTest.java#L22